glean-core/tests/dual_labeled_counter.rs

```diff
@@ -0,0 +1,527 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
```

Comment:
Regardless of which behavior we decide with the 16 categories/16 keys issue I mentioned above, we should have a test for that behavior.